src/request.cpp

```diff
@@ -122,17 +122,9 @@ set_expect_100_continue(bool b)
         return;
     }
 
+    BOOST_ASSERT(h_.md.expect.ec.failed());
     if(b)
     {
-        if(! h_.md.expect.ec.failed())
-        {
-            // remove all but one
-            raw_erase_n(
-                field::expect,
-                h_.md.expect.count - 1);
-            return;
-        }
-
         erase(field::expect);
```

Comment:
Could this be
```
raw_erase_n( field::expect, h_.md.expect.count  );
```
instead?

Reply:
This would work if we could guarantee that all Expect fields are kept adjacent in memory, which doesn't seem to be the case.

This is a good question though because it made me realize that the tests are actually over-fitted and would've hid this issue. I'm updating the tests by simply inserting a `Content-Length` header between them.

Reply:
This required a little of massaging of the metadata but I made it work and I think it's a bit cleaner than before.